conans/model/build_info.py

```diff
@@ -447,28 +447,30 @@ def set_relative_base_folder(self, folder):
                 origin = getattr(component, varname)
                 if origin is not None:
                     origin[:] = [os.path.join(folder, el) for el in origin]
-            if component._generator_properties is not None:
-                updates = {}
-                for prop_name, value in component._generator_properties.items():
-                    if prop_name == "cmake_build_modules":
-                        if isinstance(value, list):
-                            updates[prop_name] = [os.path.join(folder, v) for v in value]
-                        else:
-                            updates[prop_name] = os.path.join(folder, value)
-                component._generator_properties.update(updates)
+            properties = component._generator_properties
```

Comment:
This was a minor refactor, no need to iterate all properties, if only 1 used ( I had to do the same below)